pkg/roles/service.go

```diff
@@ -450,3 +455,215 @@ func NewService(ofga OpenFGAClientInterface, wpool pool.WorkerPoolInterface, tra
+				Relation: *e.Entitlement,
+				Object:   *e.Resource,
+			},
+		)
+	}
+
+	if err := s.core.AssignPermissions(ctx, r.ID, permissions...); err != nil {
+		return nil, v1.NewUnknownError(err.Error())
+	}
+	// ###################################
+	return role, nil
+}
+
+// GetRole returns a single Role.
+func (s *V1Service) GetRole(ctx context.Context, roleId string) (*resources.Role, error) {
+	ctx, span := s.core.tracer.Start(ctx, "roles.V1Service.GetRole")
+	defer span.End()
+
+	principal := authentication.PrincipalFromContext(ctx)
+
+	if principal == nil {
+		return nil, v1.NewAuthorizationError("unauthorized")
+	}
+	r, err := s.core.GetRole(ctx, principal.Identifier(), roleId)
+
+	if err != nil {
+		return nil, v1.NewUnknownError(err.Error())
+	}
+
+	if r == nil {
+		return nil, v1.NewNotFoundError("role not found")
+	}
+
+	role := new(resources.Role)
+
+	role.Id = &r.ID
+	role.Name = r.Name
+
+	return role, nil
+}
+
+// UpdateRole updates a Role.
+func (s *V1Service) UpdateRole(ctx context.Context, role *resources.Role) (*resources.Role, error) {
+	_, span := s.core.tracer.Start(ctx, "roles.V1Service.UpdateRole")
+	defer span.End()
+
+	return nil, v1.NewNotImplementedError("endpoint not implemented")
+}
+
+func (s *V1Service) DeleteRole(ctx context.Context, roleId string) (bool, error) {
+	ctx, span := s.core.tracer.Start(ctx, "roles.V1Service.DeleteRole")
+	defer span.End()
+
+	if err := s.core.DeleteRole(ctx, roleId); err != nil {
+		return false, v1.NewUnknownError(err.Error())
+	}
+
+	return true, nil
+}
+
+// GetRoleEntitlements returns a page of Entitlements for Role `roleId`.
+func (s *V1Service) GetRoleEntitlements(ctx context.Context, roleId string, params *resources.GetRolesItemEntitlementsParams) (*resources.PaginatedResponse[resources.EntityEntitlement], error) {
+	ctx, span := s.core.tracer.Start(ctx, "roles.V1Service.GetRoleEntitlements")
+	defer span.End()
+
+	paginator := types.NewTokenPaginator(s.core.tracer, s.core.logger)
+
+	if err := paginator.LoadFromString(ctx, *params.NextToken); err != nil {
+		s.core.logger.Error(err)
+	}
+
+	permissions, pageTokens, err := s.core.ListPermissions(ctx, roleId, paginator.GetAllTokens(ctx))
+
+	if err != nil {
+		return nil, v1.NewUnknownError(err.Error())
+	}
+
+	paginator.SetTokens(ctx, pageTokens)
+	metaParam, err := paginator.PaginationHeader(ctx)
+	if err != nil {
+		s.core.logger.Errorf("error producing pagination meta param: %s", err)
+		metaParam = ""
+	}
+
+	r := new(resources.PaginatedResponse[resources.EntityEntitlement])
+	r.Meta = resources.ResponseMeta{Size: len(permissions)}
+	r.Data = make([]resources.EntityEntitlement, 0)
+	r.Next.PageToken = &metaParam
+
+	for _, permission := range permissions {
+		p := authorization.NewURNFromURLParam(permission)
+
+		r.Data = append(
+			r.Data,
+			resources.EntityEntitlement{
+				Entitlement: p.Relation(),
+				EntityType:  strings.Split(p.Object(), ":")[0],
+				EntityId:    strings.Split(p.Object(), ":")[1],
+			},
+		)
```

Comment:
Nitpick: switch to a controlled split with `strings.SplitN` and reuse the slice

```suggestion
                entity := strings.SplitN(p.Object(), ":", 2)
		r.Data = append(
			r.Data,
			resources.EntityEntitlement{
				Entitlement: p.Relation(),
				EntityType:  entity[0],
				EntityId:       entity[1],
			},
		)
```